urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/grinder/controller/ControllerGUI.java

```diff
@@ -0,0 +1,763 @@
+		gc.gridx = 2;
+		gc.weightx = 0;
+		panel_controls.add(lbl_articleList, gc);
+
+		gc.gridx = 0;
+		gc.gridy = 1;
+		panel_controls.add(lbl_webServerAddress, gc);
+		gc.gridx = 1;
+		panel_controls.add(field_webServerAddress, gc);
+		gc.gridx = 2;
+		gc.gridheight = 3;
+		gc.gridwidth = 1;
+		panel_controls.add(lbl_articleListStatus, gc);
+
+		gc.gridx = 0;
+		gc.gridy = 2;
+		gc.gridheight = 1;
+		gc.gridwidth = 1;
+		panel_controls.add(lbl_webSocketAddress, gc);
+		gc.gridx = 1;
+		panel_controls.add(field_webSocketURI, gc);
+
+		gc.gridx = 0;
+		gc.gridy = 3;
+		panel_controls.add(lbl_startTime, gc);
+		gc.gridx = 1;
+		panel_controls.add(field_startTime, gc);
+
+		gc.gridx = 0;
+		gc.gridy = 4;
+		panel_controls.add(btn_resetStartTime, gc);
+		gc.gridx = 1;
+		panel_controls.add(btn_deployTicket, gc);
+		gc.gridx = 2;
+		gc.gridwidth = 1;
+		panel_controls.add(btn_articleList, gc);
+
+		gc.gridx = 0;
+		gc.gridy = 5;
+		gc.gridwidth = 3;
+		panel_controls.add(btn_collectLogFiles, gc);
+
+		panel_controls.setMinimumSize(new Dimension(500, 200));
+		panel_controls.setPreferredSize(new Dimension(500, 200));
+		setTitledBorder(panel_controls, "test controls");
+	}
+
+	/**
+	 * Builds the log panel.
+	 */
+	private void buildLogPanel() {
+		panel_log = new JPanel(new GridLayout());
+		area_log = new JTextArea();
+
+		area_log.setEditable(false);
+		area_log.setFont(new Font("Courier New", Font.PLAIN, 11));
+		area_log.setLineWrap(true);
+
+		// make area scrollable....
+		final JScrollPane scroll_areaLog = new JScrollPane(area_log);
+
+		panel_log.add(scroll_areaLog);
+		setTitledBorder(panel_log, "log");
+	}
+
+	/** The action deploy test ticket. */
+	private final Action action_deployTestTicket = new AbstractAction("deploy test ticket!") {
+
+		@Override
+		public void actionPerformed(final ActionEvent arg0) {
+			if (ctrl.isArticleListLoaded())
+			{
+				final boolean success = ctrl.deployTestTickets();
+				if (!success)
+				{
+					errorDialog("No clients connected", "There is not a single client connected.");
+				}
+			} else
+			{
+				errorDialog("Article list missing", "Please specify a proper article list first.");
+			}
+		}
+	};
+
+	/** The action deploy test ticket. */
+	private final Action action_collectLogs = new AbstractAction("manually collect log files") {
+
+		@Override
+		public void actionPerformed(final ActionEvent arg0) {
+			final int answer = confirmDialog("Manually collect log files",
+					"Do you really want to collect the logfiles of all finished clients now?\n" +
+					"Please take care of proper logfile documentation.");
+
+			if (answer == JOptionPane.OK_OPTION)
+			{
+				ctrl.collectClientLogFiles();
+			}
+		}
+	};
+
+	/** The action deploy test ticket. */
+	private final Action action_resetTime = new AbstractAction("set time now + 10s") {
+
+		@Override
+		public void actionPerformed(final ActionEvent arg0) {
+			final Format f = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.